### DISCUSSION TITLE
Import & require - @architect/shared in a react-app / begin project

### DISCUSSION THREAD

**Discussion by toychicken** [beginner-corp/begin-community#60] [Q&A] [answered]
I'm doing a little project in react / begin - using the default create-react-app setup from Begin.com - I have a `/shared` directory - where I'm storing some constants (object keys) that I'd like to share between the API and the React app. However, I'm running into an issue, because the API wants to be using `require` and the react app is using `import`. I'm sure there's a simple solution - but my tired lockdown brain doesn't seem to be able to work out what it is... Help is most welcome :D

**Comment by kristoferjoseph** [comment, marked as answer]
@toychicken This is a common problem and has an easy fix.
I use [esm](https://github.com/standard-things/esm) which allows you to require ES Modules in node.

**Reply by ryanblock** [reply]
For reference I'd love to see a quick code snippet if you have any handy!

**Reply by kristoferjoseph** [reply]
Sure thing.

Let's say we have an ES Module `src/views/modules/main.js` that loooks like:
```
export default function Main(props) {
  return `
    <h1>YOLO!</h1>
  `
}
```

You can require it in your lambda function by augmenting `require` with `esm` like so:
```
const imports = require('esm')(module)
const Main = imports('@architect/views/modules/main.js').default
```

**Reply by toychicken** [reply]
I meant to say, thanks for this!

**Reply by ryanblock** [reply]
No problem! Now with Node.js esm support in Lambda, this is probably no longer necessary.
